solr/test-framework/src/java/org/apache/solr/embedded/JettySolrRunner.java

```diff
@@ -378,21 +377,9 @@ private void init(int port) {
       server.addEventListener(
           new LifeCycle.Listener() {
 
-            @Override
-            public void lifeCycleStopping(LifeCycle arg0) {}
-
-            @Override
-            public synchronized void lifeCycleStopped(LifeCycle arg0) {
-              if (coreContainerProvider != null) {
-                coreContainerProvider.close();
-              }
-            }
-
-            @Override
-            public void lifeCycleStarting(LifeCycle arg0) {}
-
             @Override
             public synchronized void lifeCycleStarted(LifeCycle arg0) {
+              // awkwardly, parts of Solr want to know the port but we don't know that until now
```

Comment:
probably something we could fix.  The Jetty server inserts itself into the context, thus CoreContainerProvider should be able to pull it out on initialization in order to find the port.